### ISSUE TITLE
Allow Lint/ConstantResolution To Accept Unqualified Constant In Current Scope

### ISSUE THREAD

**Issue by jfelchner** [rubocop/rubocop#9284]
## Is your feature request related to a problem? Please describe.

First I want to say that I _love_ this cop and I think that enabling it will solve a lot of _very_ hard to track down bugs, but I have one thing that I think would drastically improve it:

I don't even know if this is possible with the parser, but I think that if a constant is _directly_ defined in the current scope, there should be no reason to fully qualify it.  To do so just increases verbosity with no practical benefit.

### Example 1

```ruby
module Bar
  OUTER_CONSTANT = "peas_and_carrots".freeze

  class Foo
    THIS_CONSTANT_I_HAVE = "trivial".freeze

    def i_love_constants
      [
        THIS_CONSTANT_I_HAVE.upcase,  # This doesn't need qualified because it's on the current scope
        ::Bar::OUTER_CONSTANT.upcase, # This is qualified because it's defined a level up
      ]
        .join('-')
    end

    def convoluted_class_reference_for_some_reason_instead_of_calling_self_class
      Foo.new # This is not qualified because it's literally the current class
    end
  end
end
```

### Example 2

```ruby
# RSpec

module Bar
  describe Foo do
    it "does a thing" do
      # Here `Foo` is not qualified because it's the current scope which is one
      # of the benefits of this style of spec layout.
      expect(Foo.new.i_love_constants).to eql "TRIVIAL-PEAS_AND_CARROTS"
    end
  end
end
```

For the above examples, I would expect that the cop would not register any offenses.

**Comment by bbatsov** [comment]
Closing as stale — no engagement in over 5 years. Feel free to reopen if this is still needed.
